### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +29,35 @@ public Integer getLength() {
...
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        final int prime = 31;
+        result = prime * result + Objects.hashCode(width);
+        result = prime * result + Objects.hashCode(length);
+        result = prime * result + Objects.hashCode(color);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_06.html check about using Objects

Comment (vremere-a) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,37 @@ public Integer getLength() {
...
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj instanceof Rectangle) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: what the difference between instanceof and getClass()?

Comment (vremere-a) [reply src/main/java/core/basesyntax/Rectangle.java]: instanceof use when we use extends

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**